### PR TITLE
Use non-blocking function for getting authenticated user

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Replaces Rx based `currentUser` to a coroutine based method.

Verified the changes by running the app locally and performing signin/signout

@gino-m @scolsen   PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt [hunk 1]
```diff
@@ -49,26 +49,22 @@ internal constructor(
   private val userRepository: UserRepository
 ) : AbstractViewModel() {
 
+  // TODO: Expose non-mutable state flow.
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt:52]: Please add bug nos to new TODOs here and elsewhere.

**Review by gino-m — CHANGES_REQUESTED** [review]
A few nits.

File: ground/src/main/java/com/google/android/ground/system/auth/AuthenticationManager.kt [hunk 2]
```diff
@@ -17,12 +17,27 @@ package com.google.android.ground.system.auth
...
+
   fun signOut()
+
+  /** Returns the logged-in user. */
+  suspend fun getAuthenticatedUser(): User =
+    signInState
+      .asFlow()
+      .filter { it.state == SignInState.State.SIGNED_IN }
+      .map { it.result.getOrNull() }
+      .filterNotNull()
```
[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/system/auth/AuthenticationManager.kt]: Done

File: ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt]: Done

**Review by gino-m — APPROVED** [review]

**Pull request merged**